common/buildcraft/core/client/RenderTickListener.java

```diff
@@ -106,56 +110,33 @@ public static void renderOverlay(RenderGameOverlayEvent.Text event) {
         if (mouseOver == null) {
             return;
         }
-        boolean both = BCCoreConfig.useLocalServerOnClient;
-
-        IDebuggable client = getDebuggableObject(mouseOver);
-        IDebuggable server = both ? getServer(client) : null;
-
-        if (client == null) return;
-        EnumFacing side = mouseOver.sideHit;
-        if (server == null) {
-            client.getDebugInfo(left, right, side);
-        } else {
-            List<String> serverLeft = new ArrayList<>();
-            List<String> serverRight = new ArrayList<>();
 
+        getDebuggableObject(mouseOver).ifPresent(client -> {
+            EnumFacing side = mouseOver.sideHit;
+            MessageManager.sendToServer(new MessageDebuggableRequest(client.getPos(), side));
             List<String> clientLeft = new ArrayList<>();
             List<String> clientRight = new ArrayList<>();
-
-            server.getDebugInfo(serverLeft, serverRight, side);
-            client.getDebugInfo(clientLeft, clientRight, side);
+            client.getClientDebugInfo(clientLeft, clientRight, side);
 
             final String headerFirst = DIFF_HEADER_FORMATTING + "SERVER:";
             final String headerSecond = DIFF_HEADER_FORMATTING + "CLIENT:";
-            appendDiff(left, serverLeft, clientLeft, headerFirst, headerSecond);
-            appendDiff(right, serverRight, clientRight, headerFirst, headerSecond);
-
-        }
+            appendDiff(left, ClientDebuggables.SERVER_LEFT, clientLeft, headerFirst, headerSecond);
+            appendDiff(right, ClientDebuggables.SERVER_RIGHT, clientRight, headerFirst, headerSecond);
+        });
     }
 
-    private static IDebuggable getDebuggableObject(RayTraceResult mouseOver) {
+    private static <T extends TileBC_Neptune & IDebuggable> Optional<T> getDebuggableObject(RayTraceResult mouseOver) {
```

Comment:
Don't mandate TileBC_Neptune (was that a typo? You never checked for TileBC_Neptune in an instanceof check). In the future this might also be entities as well.

Reply:
I rewrote code a few times, one of implementations was inside TileBC_Neptune.